src/makeCalculator.js

```diff
@@ -37,7 +37,48 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  const calculator = {
+    add: (number) => {
+      result += number;
+
+      return calculator;
+    },
+    subtract: (number) => {
+      result -= number;
+
+      return calculator;
+    },
+    multiply: (number) => {
+      result *= number;
+
+      return calculator;
+    },
+    divide: (number) => {
+      if (result === 0) {
+        return 'You cannot divide by zero';
+      }
+      result /= number;
+
+      return calculator;
+    },
+    reset: () => {
+      result = 0;
+
+      return calculator;
+    },
+    operate: (callback, number) => {
+      callback(number);
```

Comment:
The 'operate' function does not affect the 'result' value. It should be implemented accordingly so as to operate on the result.